### DISCUSSION TITLE
Display warning when user uploads assets that don’t fit field conditions

### DISCUSSION THREAD

**Discussion by ReedBe** [craftcms/cms#13729] [Ideas]
Hello,

Ever since Craft introduced **Selectable Assets Conditions**, we’ve been using them to ensure the quality of the client-side pages. The most common use case for us is image sizes: we now always require the selected images to meet specific minimum dimensions so that our clients can’t choose images that are way too small for their intended purposes.

This has worked really well so far, but it has also created a rather annoying side effect:
At some point, pretty much every single one of our clients finds themselves in a situation where they directly upload an image that is too small into an `Assets` fields and are confused because it doesn’t appear in the already opened selection modal.

In other words, Craft allows the initial direct upload to take place but doesn’t provide a feedback to the user if it doesn’t meet the set requirements, which results in confusion (and, ultimately, a phone call or an email sent to us).

We think a rather simple solution to this would be to still display the non-selectable assets in the selection modal, but greyed-out with a tooltip that provides an explanation as to why they can’t be selected. A more explicit approach could also be to directly display a warning upon upload.

This behavior could also be optional and only happen if a new _“Show non-selectable files”_ option is checked in the field settings, in the same way _“Show unpermitted files”_ already shows files the user doesn’t have permission to view.

**Comment by brandonkelly** [comment]
Agree, that seems like a good solution.
